apps/mobile/src/navigation/RootStackNavigator.tsx

```diff
@@ -25,8 +32,20 @@ export function RootStackNavigator() {
     {}
   );
 
+  const track = useTrack();
   const isLoggedIn = query.viewer?.__typename === 'Viewer';
 
+  useEffect(() => {
+    const unsubscribe = navigationContainerRef.addListener('state', () => {
+      track('Page View', {
+        page: navigationContainerRef.getCurrentRoute()?.name,
+        params: navigationContainerRef.getCurrentRoute()?.params,
+      });
+    });
+
+    return unsubscribe;
+  }, [navigationContainerRef, track]);
+
```

Comment:
I think instead of passing the ref down, we could use `useNavigationState` for this.

```ts
const { page, params } = useNavigationState(state => {
  return {
    ... stuff here
  }
});

useEffect(()=> {
  track here
}, [page, params]);
```

Reply:
something like that, I'm pretty sure we don't have to pass a ref down though.